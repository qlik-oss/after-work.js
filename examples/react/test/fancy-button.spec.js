/* global aw */
/* eslint import/no-unresolved: 0 */

import React from 'react';
import renderer from 'react-test-renderer';
import FancyButton from '../src/fancy-button';

describe('FancyButton', () => {
  it('renders fancy with button as span or div', () => {
    const [{ default: FancySpan }] = aw.mock(
      [['**/react/src/button.jsx', () => () => <span>hhhhh</span>]],
      ['../src/fancy-button'],
    );
    const [{ default: FancyDiv }] = aw.mock(
      [
        ['**/react/src/button.jsx', () => () => <div>ggggg</div>],
        ['**/muppet', () => {}],
      ],
      ['../src/fancy-button'],
    );
    const tree1 = renderer.create(<FancySpan>span</FancySpan>).toJSON();
    const tree2 = renderer.create(<FancyDiv>div</FancyDiv>).toJSON();
    expect(tree1).toMatchSnapshot();
    expect(tree2).toMatchSnapshot();
  });

  it('renders fancy with button as is', () => {
    const [{ default: FancyPancy }] = aw.mock([], ['../src/fancy-button']);
    const tree1 = renderer.create(<FancyPancy>pancy</FancyPancy>).toJSON();
    expect(tree1).toMatchSnapshot();
  });

  it('renders fancy with button as is from import', () => {
    const tree1 = renderer.create(<FancyButton>fancy1</FancyButton>).toJSON();
    expect(tree1).toMatchSnapshot();
  });

  it('renders fancy with button mock from file', () => {
    const [{ default: FancyFileMock }] = aw.mock(
      [
        ['**/react/src/button.jsx', './examples/react/test/button-mock.js'],
        ['foo', () => {}],
      ],
      ['../src/fancy-button'],
    );
    const tree1 = renderer
      .create(<FancyFileMock>file mock</FancyFileMock>)
      .toJSON();
    expect(tree1).toMatchSnapshot();
  });
});
