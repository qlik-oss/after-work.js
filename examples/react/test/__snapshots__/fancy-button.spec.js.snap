// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`'renders fancy with button as is from import' 1`] = `
<div>
  <button>
    fancy1
  </button>
</div>
`;

exports[`'renders fancy with button as is' 1`] = `
<div>
  <button>
    pancy
  </button>
</div>
`;

exports[`'renders fancy with button as span or div' 1`] = `
<div>
  <span>
    hhhhh
  </span>
</div>
`;

exports[`'renders fancy with button as span or div' 2`] = `
<div>
  <div>
    ggggg
  </div>
</div>
`;

exports[`'renders fancy with button mock from file' 1`] = `
<div>
  <span>
    file mock
  </span>
</div>
`;
